x-pack/plugins/lens/public/indexpattern_datasource/field_item.tsx

```diff
@@ -432,7 +432,7 @@ function FieldItemPopoverContents(props: State & FieldItemProps) {
 
   let formatter: { convert: (data: unknown) => string };
   if (indexPattern.fieldFormatMap && indexPattern.fieldFormatMap[field.name]) {
-    const FormatType = fieldFormats.getType(indexPattern.fieldFormatMap[field.name].id);
+    const FormatType = fieldFormats.getType(indexPattern.fieldFormatMap[field.name].id as string);
```

Comment:
Perhaps `fieldFormats.getType` needs better type info instead of using `as string` on the id

Reply:
Looks like the object could come from user input, and the ID is not validated to be present: https://github.com/elastic/kibana/blob/8.3/src/plugins/data_views/server/rest_api_routes/util/schemas.ts#L13.  

There is quite a lot of test code that relies on these fields being optional as well.